programs/mpl-core/src/plugins/plugin_registry.rs

```diff
@@ -70,12 +91,18 @@ impl RegistryRecord {
     }
 }
 
-/// A simple type to store the mapping of external Plugin authority to Plugin data.
+/// A type to store the mapping of third party plugin type to third party plugin header and data.
 #[repr(C)]
 #[derive(Clone, BorshSerialize, BorshDeserialize, Debug)]
 pub struct ExternalPluginRecord {
+    /// The third party plugin key.
+    pub plugin_key: ExternalPluginKey,
     /// The authority of the external plugin.
     pub authority: Authority,
-    /// The offset to the plugin in the account.
-    pub offset: usize,
+    /// The lifecyle events for which the the third party plugin is active.
+    pub lifecycle_checks: Option<HashMap<LifecycleEvent, ExternalCheckResult>>,
```

Comment:
Yeah I had changed it from Vec to HashMap since it was so much nicer to deal with in Rust that way... but I think you are right and its a mistake for CUs to use a HashMap here.  Will switch it back to Vec and just iterate through it.

Reply:
Switched it back to Vec.